src/libraries/System.IO.FileSystem/tests/File/GetSetTimes_SafeFileHandle.cs

```diff
@@ -0,0 +1,93 @@
+            path,
+            FileMode.OpenOrCreate,
+            FileAccess.ReadWrite,
+            FileShare.ReadWrite);
+
+        protected override void SetCreationTime(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetCreationTime(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetCreationTime(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetCreationTime(fileHandle);
+        }
+
+        protected override void SetCreationTimeUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetCreationTimeUtc(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetCreationTimeUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetCreationTimeUtc(fileHandle);
+        }
+
+        protected override void SetLastAccessTime(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastAccessTime(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetLastAccessTime(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastAccessTime(fileHandle);
+        }
+
+        protected override void SetLastAccessTimeUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastAccessTimeUtc(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetLastAccessTimeUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastAccessTimeUtc(fileHandle);
+        }
+
+        protected override void SetLastWriteTime(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastWriteTime(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetLastWriteTime(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastWriteTime(fileHandle);
+        }
+
+        protected override void SetLastWriteTimeUtc(string path, DateTime creationTime)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            File.SetLastWriteTimeUtc(fileHandle, creationTime);
+        }
+
+        protected override DateTime GetLastWriteTimeUtc(string path)
+        {
+            using var fileHandle = OpenFileHandle(path);
+            return File.GetLastWriteTimeUtc(fileHandle);
+        }
+
+        protected override void SettingUpdatesPropertiesCore(string item)
+        {
+            // Override `SettingUpdatesPropertiesCore`, because SafeFileHandles are not able to do this
```

Comment:
This shortcuts some tests. What are SafeFileHandles not able to do? Can you give an example?

Reply:
As far as I can remember I always encountered here an AccessViolationException. I will investigate this in more detail tomorrow.

Reply:
> It is an UnauthorizedAccessException (not AccessViolationException as I thought). I am not sure if this is the intended behavior.

`SettingUpdatesPropertiesWhenReadOnly` creates a read-only file, and you are trying to open it `FileAccess.ReadWrite`, so it throws `UnauthorizedAccessException`.

If you change to `FileAccess.Read` it should pass.

> I also noticed that in the symlink case the date is not changed. Is this intentional? If not, do you have a possible solution?

With the `SafeHandle` APIs, the target of the symlink is changed and not the link. So this test is expected to fail.

Can you rename the test to `SettingPropertiesOnSymlink`, add a `virtual bool SettingPropertiesUpdatesLink` which is set to `true` in the base, and `false` in the `SafeHandle` derived class, and update the test expectation based on this `bool`.